core/src/main/c/freebsd/kqueue.c

```diff
@@ -89,5 +90,35 @@ JNIEXPORT jint JNICALL Java_io_questdb_network_KqueueAccessor_kevent
                          &dontBlock);
 }
 
+JNIEXPORT jlong JNICALL Java_io_questdb_network_KqueueAccessor_pipe
+        (JNIEnv *e, jclass cl) {
+    int fds[2];
+    int res = pipe2(fds, O_NONBLOCK);
+    if (res < 0) {
+        return res;
+    }
+    return (jlong) fds[0] << 32 | (jlong) fds[1];
+}
 
+JNIEXPORT jint JNICALL Java_io_questdb_network_KqueueAccessor_readPipe
+        (JNIEnv *e, jclass cl, jint fd) {
+    char buf[1];
+    ssize_t s;
+    s = read((int) fd, buf, 1);
```

Comment:
nit: this is something @amunra pointed out on one of my PRs. Apparently it is better to use this syntax:
```suggestion
    s = read((int) fd, &buf[0], 1);
```

Reply:
Done in 1f11af7c55d62ed350bfdba5dffd5370f4fc17c8